scripts/build.js

```diff
@@ -0,0 +1,161 @@
+/*!
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * Developer Documentation
+ *
+ * This script should be ran anytime new code is introduced (PRs, merges, etc.)
+ * Here's a breakdown of what happens (and when)
+ *
+ * - First we check to see if any modules have had code updates. A code update
+ *   is either a .js file update or a dependency update. In the event that a
+ *   module did not receive a code update (could be a markdown update), we'll
+ *   exit early.
+ *
+ * - For each module that has updates, we'll install their dependencies and run
+ *   their unit tests.
+ *
+ * - Assuming all the unit tests pass, we'll then attempt to get code coverage
+ *   for the changed files.
+ *
+ * - At this point, if this "build" was triggered by a PR or is a merge to a
+ *   branch other than master, we'll exit early here. All steps after this point
+ *   assume a merge to the master branch.
+ *
+ * - Next we'll run system tests for any and all modules that have recieved code
+ *   updates.
+ *
+ * - Assuming all system tests pass, we'll create symlinks for each module
+ *   via `npm link`. We'll then search for any modules that have the updated
+ *   modules listed as dependencies or devDependencies. For each dependent
```

Comment:
Yeah, and I think that's implied, since if we're messing with common, that means we need it for the benefit of another module. IOW, we're not just playing around in there.